org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/jdt/JdtSourceLookUpProvider.java

```diff
@@ -63,22 +62,27 @@ public void initialize(Map<String, Object> props) {
         }
 
         String[] fqns = new String[lines.length];
-        Path sourcePath = Paths.get(sourceFilePath);
-        ICompilationUnit cu = JDTUtils.resolveCompilationUnit(sourcePath.toUri());
-        
+        ITypeRoot typeRoot = JDTUtils.resolveCompilationUnit(uri);
+        if (typeRoot == null) {
+            typeRoot = JDTUtils.resolveClassFile(uri);
+        }
+
         for (int i = 0; i < lines.length; i++) {
             String fqn = null;
-            try {
-                int offset = JsonRpcHelpers.toOffset(cu.getBuffer(), lines[i], columns[i]);
-                IJavaElement javaElement = cu.getElementAt(offset);
-                if (javaElement instanceof SourceField || javaElement instanceof SourceMethod) {
-                    IType type = ((IMember) javaElement).getDeclaringType();
-                    fqn = type.getFullyQualifiedName();
-                } else if (javaElement instanceof SourceType) {
-                    fqn = ((SourceType) javaElement).getFullyQualifiedName();
+            if (typeRoot != null) {
+                try {
+                    int offset = JsonRpcHelpers.toOffset(typeRoot.getBuffer(), lines[i], columns[i]);
+                    IJavaElement javaElement = typeRoot.getElementAt(offset);
+                    if (javaElement instanceof SourceField || javaElement instanceof SourceMethod
+                            || javaElement instanceof BinaryMember) {
+                        IType type = ((IMember) javaElement).getDeclaringType();
+                        fqn = type.getFullyQualifiedName();
+                    } else if (javaElement instanceof SourceType) {
+                        fqn = ((SourceType) javaElement).getFullyQualifiedName();
+                    }
+                } catch (JavaModelException e) {
+                    Logger.logException("Add breakpoint exception", e);
```

Comment:
Why should the getFullyQualifiedName utility function throw breakpoint exception?